app/controllers/api/admin_controller.rb

```diff
@@ -0,0 +1,85 @@
+class Api::AdminController < Api::ApplicationController
+  before_action :admin_user
+
+  def view_profile
+    @user = User.find(params[:id])
+
+    render json: @user, serializer: UserSerializer
+  end
+
+  def view_users
+    @users = User.all
```

Comment:
There are multiple places in the API controllers where you instantiate global variables (`@users` instead of `usaers`). These variables are used only in this method, so they should be private

Reply:
Also, this could be a 1-liner: `render json: User.all` (you also optionally add `, each_serializer: UserSerializer` in case it doesn't use it by default)

Reply:
Please adapt these to all requests